### ISSUE TITLE
Client side export doesn't work

### ISSUE THREAD

**Issue by joh024** [jbkunst/highcharter#126]
I've been trying to add [client-side export](http://www.highcharts.com/docs/export-module/client-side-export), but no matter what I try the buttons always lead to the export url.

I've tried:

* Just including the `offline-exporting.js` script in `tags$head`:
```R
tags$script(src = "http://code.highcharts.com/modules/offline-exporting.js")
```

* Including both the `exporting.js` and the `offline-exporting.js` script in `tags$head` (per the [jsFiddle](http://jsfiddle.net/gh/get/jquery/1.11.0/highslide-software/highcharts.com/tree/master/samples/highcharts/exporting/offline-download/) example):
```R
tags$script(src = "http://code.highcharts.com/modules/exporting.js"),
tags$script(src = "http://code.highcharts.com/modules/offline-exporting.js")
```

* Setting `fallbackToExportServer = FALSE` in `hc_exporting` - still leads to going to the export server.

I _can_ change the export url (by setting the `url` argument in `hc_exporting` to something else), but that doesn't help with getting client side exporting to work.

The fact that `fallbackToExportServer = FALSE` doesn't seem to work seems to imply that there might be a bug in how exporting is handled, which also leads to the client side export from failing?

**Comment by jbkunst** [comment]
Mmm... What about the console? Give some error?

**Comment by joh024** [comment]
No errors reported in R or on the web console.

**Comment by joh024** [comment]
Possibly related information, it seems that the package is using Highcharts v4.2.3, while the scripts linked in the example are for v4.2.5.

Using the linked v4.2.5 `exporting.js` I just encountered an obscure problem, that was resolved by using the version that comes with the package.

**Comment by joh024** [comment]
Alas, still doesn't work even when I use the [matching version of offline-exporting.js](http://code.highcharts.com/4.2.3/modules/offline-exporting.js).

**Comment by jbkunst** [comment]
Uff, ok, I'll take some time to see this.

I'll let you know if I discover something

**Comment by joh024** [comment]
Using the API directly [`.exportChartLocal()`](http://api.highcharts.com/highcharts#Chart.exportChartLocal) works, so it might be an issue with how the navigation buttons are implemented in highcharter.

**Comment by jbkunst** [comment]
Hi! 

Can you give me a reproducible example? I'll test with `highchart2()` function which use less plugins thant the `highchart()`. May be some pluging is making noise.

**Comment by jbkunst** [comment]
@joh024 

I modifed the yaml file to add exporting and offline exporting scripts, then I tested with firefox and it works, can you check and tell me what happend please? Thanks!

**Comment by joh024** [comment]
Sorry for the slow reply.

I can confirm that it also works for me, but this solution forces offline exporting, which I don't think is ideal.

The ideal is that it defaults to the external server, but with the option to include the offline exporting script if desired.

Failing that, as using the API directly worked previously, I think it's better to reverse this change to not force offline exporting.

**Comment by jbkunst** [comment]
Mmm... I think you're right. I'll take some times/tries to see how can we can fix this

Thanks for your thoughts

**Comment by schmockolaus** [comment]
I also agree with joh024. It would be really great if you could choose between offline exporting and an external server.

**Comment by joh024** [comment]
I see that the current version again seems to be forcing offline exporting, and for whatever reason even if I use code to force non-local exporting (using `.exportChart`, rather than `.exportChartLocal`) it seems to fail (possibly because the offline-exporting code seems to replace a lot of the code).

Again, it would be much better if the default is external server (for greater browser compatibility), with the **option** of including the offline exporting script.

**Comment by jbkunst** [comment]
Let me check

**Comment by Cstaf** [comment]
I'm wondering if there ever was a resolution to this as well. I'm currently trying to implement a client-side export in a Shiny-app but no matter what i do it's directed to 'https://export.highcharts.com/'.

**Comment by MatveevEL** [comment]
Hi!
As I understand the problem with the export is still there. This reproducible example
Example with forced include exportChartLocal
`export <- list(
    list(text = "PNG",
         onclick = JS("function () {this.exportChartLocal();  }")),
    list(text = "JPEG",
         onclick = JS("function () {this.exportChartLocal({ type: 'image/jpeg' }); }"))
)`

`highchart() %>%
    hc_chart(type = "line" ) %>%
    hc_xAxis(cars$speed) %>%
    hc_add_series(cars$dist) %>%
    hc_exporting(enabled = TRUE,  allowHTML  = F, fallbackToExportServer  = F, 
                 formAttributes = list(target = "_blank"),
                 buttons = list(contextButton = list(menuItems = export)))`

Or I do something wrong?

SessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
other attached packages:
 [1] V8_1.5               broom_0.3.7          MASS_7.3-49          nloptr_1.2.1         foreach_1.5.0        stringr_1.3.1       
 [7] tidyr_0.8.3          highcharter_0.7.0    dplyr_0.8.0.1        xlsx_0.6.1           RevoUtils_11.0.0     RevoUtilsMath_11.0.0
